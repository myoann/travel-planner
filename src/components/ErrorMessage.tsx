export default function ErrorMessage({ message }: { message: string }) {
  return <div className="px-8 text-red-500 text-lg">{message}</div>;
}
