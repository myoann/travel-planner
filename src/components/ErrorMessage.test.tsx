import { render, screen } from "@testing-library/react";

import ErrorMessage from "./ErrorMessage";

const defaultProps = {
  message: "This is an error message",
};

describe("ErrorMessage", () => {
  it("matches the snapshot", () => {
    const { container } = render(<ErrorMessage {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });

  it("renders the given error message props", () => {
    render(<ErrorMessage {...defaultProps} />);

    const errorMessage = screen.getByText(defaultProps.message);
    expect(errorMessage).toBeInTheDocument();
  });
});
