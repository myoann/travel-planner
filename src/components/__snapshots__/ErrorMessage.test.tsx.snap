// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage matches the snapshot 1`] = `
<div>
  <div
    class="px-8 text-red-500 text-lg"
  >
    This is an error message
  </div>
</div>
`;
